src/main/java/com/gmail/val59000mc/players/PlayersManager.java

```diff
@@ -426,19 +427,47 @@ public void setAllPlayersEndGame() {
 		Bukkit.getServer().getPluginManager().callEvent(event);
 
 		double reward = cfg.getRewardWinEnvent();
+		List<String> winCommands = cfg.getWinCommands();
+		boolean cmdsNeedsPlayer = false;
+		for (String cmd : winCommands){
+			if (cmd.contains("%name%")){
+				cmdsNeedsPlayer = true;
+				break;
```

Comment:
Don't really understand this ``cmdsNeedsPlayer``, if one of the commands needs a play they all do?

Reply:
No need to run the command for winners.size() times if the command does not take the %name% placeholders, if it has the %name% placeholder, then it's because the command is specific to every player. If we run the command in the 
`for UhcPlayer player : winners`, then the command will be run a lot of times for nothing

Reply:
I understand that but that's not what your code does. Your code checks if any of the commands contains %name% in which case it executes all commands for all winners.

Reply:
You'll still have to remove the break. This way the winCommandsPlayer list will at most contain 1 command.